### PR TITLE
Use SlugField on Form for  Computed Field Slug.

### PR DESCRIPTION
**Pull request by smk4664**
Also moves the Slug field below Label.

### Fixes: #683 

Set ComputedFieldForm to use SlugField.

### CODE REVIEW

**Comment by smk4664** [comment]
Marking as WIP while I investigate django-extensions AutoSlugField.

File: nautobot/core/settings.py [hunk 1]
```diff
@@ -303,6 +303,7 @@
     "health_check.db",
     "health_check.cache",
     "health_check.storage",
+    "django_extensions",
```
Comment (glennmatthews) [inline nautobot/core/settings.py:306]: Just checking to confirm that this is the appropriate position in the list to inject django_extensions. In particular I see that django-extensions has a `runjob` management command which would collide with nautobot's own `runjob` command - can you confirm that `nautobot-server runjob` still correctly uses the nautobot command and not the django-extensions command?

File: nautobot/dcim/models/devices.py [hunk 2]
```diff
@@ -51,7 +52,7 @@ class Manufacturer(OrganizationalModel):
     """
 
     name = models.CharField(max_length=100, unique=True)
-    slug = models.SlugField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name", max_length=100, unique=True, editable=True, overwrite_on_add=False)
```
Comment (glennmatthews) [inline nautobot/dcim/models/devices.py:55]: What do you think about creating our own custom subclass of `AutoSlugField` in `nautobot.utilities.fields` that would have all of these default values as presets? Would reduce the amount of code we have to repeat all over the place and make it easier to ensure that we're defining all of our slug fields consistently across the board.


```python
class AutoSlugField(django_extensions.db.fields.AutoSlugField):

    def __init__(self, *args, **kwargs):
        kwargs.setdefault("max_length", 100)
        kwargs.setdefault("unique", True)
        kwargs.setdefault("editable", True)
        kwargs.setdefault("overwrite_on_add", False)
        super().__init__(*args, **kwargs)
```

File: nautobot/dcim/models/devices.py [hunk 3]
```diff
@@ -96,7 +97,7 @@ class DeviceType(PrimaryModel):
 
     manufacturer = models.ForeignKey(to="dcim.Manufacturer", on_delete=models.PROTECT, related_name="device_types")
     model = models.CharField(max_length=100)
-    slug = models.SlugField(max_length=100)
+    slug = AutoSlugField(populate_from="model", max_length=100, editable=True, overwrite_on_add=False)
```
Comment (glennmatthews) [inline nautobot/dcim/models/devices.py:100]: Ew, we don't require DeviceType.slug to be globally unique? That makes me sad. Good of you to preserve the existing behavior here though.

File: nautobot/dcim/models/racks.py [hunk 4]
```diff
@@ -52,7 +53,7 @@ class RackGroup(MPTTModel, OrganizationalModel):
     """
 
     name = models.CharField(max_length=100)
-    slug = models.SlugField(max_length=100)
+    slug = AutoSlugField(populate_from="name", max_length=100, editable=True, overwrite_on_add=False)
```
Comment (glennmatthews) [inline nautobot/dcim/models/racks.py:56]: Same :sad: here about not being globally unique, but correct to not change the behavior at this time.

File: nautobot/extras/models/models.py [hunk 5]
```diff
@@ -629,7 +630,7 @@ class ConfigContextSchema(OrganizationalModel):
 
     name = models.CharField(max_length=200, unique=True)
     description = models.CharField(max_length=200, blank=True)
-    slug = models.SlugField()
+    slug = AutoSlugField(populate_from="name", editable=True, overwrite_on_add=False)
```
Comment (glennmatthews) [inline nautobot/extras/models/models.py:633]: Config context schemas are a new enough feature that I'd be tempted to add `unique=True` to this one - IMHO that's an oversight.

File: nautobot/ipam/models.py [hunk 6]
```diff
@@ -976,7 +977,7 @@ class VLANGroup(OrganizationalModel):
     """
 
     name = models.CharField(max_length=100)
-    slug = models.SlugField(max_length=100)
+    slug = AutoSlugField(populate_from="name", max_length=100, editable=True, overwrite_on_add=False)
```
Comment (glennmatthews) [inline nautobot/ipam/models.py:980]: :sad: about non-globally-unique slug. I may open an issue about these.

File: nautobot/dcim/models/devices.py (hunk 3 above)
Comment (jathanism) [reply nautobot/dcim/models/devices.py:100]: Ugh.

File: nautobot/core/settings.py (hunk 1 above)
Comment (smk4664) [reply nautobot/core/settings.py:306]: I will  check, but order for management commands is first come, first serve.  Which is why we have nautobot.extras before django_rq.

File: nautobot/dcim/models/devices.py (hunk 2 above)
Comment (smk4664) [reply nautobot/dcim/models/devices.py:55]: I like that, and will make these changes shortly.

File: nautobot/dcim/models/devices.py (hunk 3 above)
Comment (smk4664) [reply nautobot/dcim/models/devices.py:100]: It uses the Meta unique_together, which caused me a ton of pain on the API front.

File: nautobot/dcim/models/racks.py (hunk 4 above)
Comment (smk4664) [reply nautobot/dcim/models/racks.py:56]: unique_together meta (site+name and site+slug)

File: nautobot/extras/models/models.py (hunk 5 above)
Comment (smk4664) [reply nautobot/extras/models/models.py:633]: Sure, I just wanted to make sure the API didn't change.

File: nautobot/ipam/models.py (hunk 6 above)
Comment (smk4664) [reply nautobot/ipam/models.py:980]: Yeah, this one I don't think uses unique_together, unlike rack and devicetype.

**Review by glennmatthews — APPROVED** [review]
Thanks for sorting out this PR vs #757. :-)

**Review by jathanism — APPROVED** [review]

**Pull request merged**